Update KvbAppFilter and TRS to Handle Event Groups

shgandhi opened the pull request:
This PR updates KvbAppFilter and TRS to handle event groups. Please see individual commits for more detail.
Note - This PR does not include the corresponding unit tests  yet.

robem reviewed (approved):
Thanks for working on this 🚀  and the commit messages 💪  Looks good to me. Some nits. There are some more `const` references possible that I didn't flag.
Did you do any manual runtime tests or is that not possible yet?

shgandhi commented on kvbc/src/kvbc_app_filter/kvbc_app_filter.cpp:
```diff
@@ -154,24 +188,60 @@ void KvbAppFilter::readBlockRange(BlockId block_id_start,
+      std::stringstream msg;
+      msg << "Couldn't retrieve event group for event_group_id: " << event_group_id;
+      throw KvbReadError(msg.str());
+    }
```
Thanks for catching, updated the message to reflect that.

shgandhi commented on thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp:
```diff
@@ -210,32 +210,50 @@ class ThinReplicaImpl {
-      LOG_ERROR(logger_, error.what());
-      config_->subscriber_list.removeBuffer(live_updates);
-      live_updates->removeAllUpdates();
+      bool event_group_enabled = true;
```
Removed the flag.

shgandhi commented:
> Thanks for working on this 🚀 and the commit messages 💪 Looks good to me. Some nits. There are some more `const` references possible that I didn't flag.
> Did you do any manual runtime tests or is that not possible yet?

Thanks for the review @robem. I have updated the PR acc. to your comments and  added the TRS changes as well to this PR, so would require your review again. I will now add the unit tests and update the PR acc. if required.

robem reviewed (approved):
I see you chose to duplicate the existing code and adapt it for EventGroups. The change itself is clear to me and I'm ok with this solution but we should do some refactoring in the near future otherwise this might become a maintenance nightmare. Well, the *_impl.hpp wasn't too pretty to begin with :)

💪  Great job so far and thanks for the hard work!

shgandhi commented:
Thanks for the review @robem. I agree we will need to refactor the code soon.
I added another commit that adds missing header guards in db_categories.h.
